### ISSUE TITLE
Property Sources from config client not available to EnvironmentPostProcessor

### ISSUE THREAD

**Issue by edeandrea** [spring-cloud/spring-cloud-config#1149]
I'm not sure whether this is an issue with Spring Boot or Spring Cloud Config, so I'm going to post to both to try and get an answer and/or this resolved (see spring-projects/spring-boot#14607).

Spring Boot Version: `2.0.4.RELEASE` (same issue with `2.0.5.RELEASE` - haven't tested with prior versions)
Spring Cloud Config Client: `2.0.1.RELEASE`

When building a Spring Boot application and building an `EnvironmentPostProcessor` (& appropriately registering it within `spring.factories`) I'm noticing that properties coming from a `configService` property source are not available when the `EnvironmentPostProcessor` runs. When I'm stepping through in the debugger I don’t see the `configService` property source in the `ConfigurableEnvironment.getPropertySources()`.

Is this a bug? Is it by design? If its a bug, is the bug with spring-cloud-config or spring boot? I was expecting all properties to have been resolved from all sources so that an `EnvironmentPostProcessor` would have access to all properties to then make decisions.

**Comment by ryanjbaxter** [comment]
Seems like @wilkinsona clarified your usage of a `EnvironmentPostProcessor`, I dont think there is anything we need to do here.  Please correct me if I am wrong.

**Comment by edeandrea** [comment]
Thank you @ryanjbaxter for responding. In the long term I guess based on @wilkinsona's comments I'm going to have to go back to the drawing board as to how to best accomplish what I am trying to do. I think I got the idea for the way I was doing it back when I was using Boot 1.x and I found an example online somewhere and it seemed like a good idea. As @wilkinsona suggested I'll post to Gitter/StackOverflow and try to solicit opinions as to what the best approach is for what I'm trying to do.

In the short term if you know if there is a workaround that would allow me to accomplish what I'm trying to do within an `EnvironmentPostProcessor` then I'm all ears.

Thanks again for the input.
